apis/hive/v1/aws/machinepool.go

```diff
@@ -21,6 +21,10 @@ type MachinePoolPlatform struct {
 	// SpotMarketOptions allows users to configure instances to be run using AWS Spot instances.
 	// +optional
 	SpotMarketOptions *SpotMarketOptions `json:"spotMarketOptions,omitempty"`
+
+	// EC2MetadataOptions defines metadata service interaction options for EC2 instances in the machine pool.
+	// +optional
+	EC2Metadata *EC2Metadata `json:"metadataService,omitempty"`
```

Comment:
Weird json name, but I see you're just matching the one in the installer type :shrug:

Reply:
Yeah, not mine.  I think better to match to installer's